google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/cloud/standard/AppEngineStandardStagingPropertiesPanel.java

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright 2017 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.tools.intellij.appengine.cloud.standard;
+
+import com.google.cloud.tools.intellij.appengine.cloud.AppEngineDeploymentConfiguration;
+import java.awt.Component;
+import javax.swing.JCheckBox;
+import javax.swing.JPanel;
+import javax.swing.JTextField;
+import org.jetbrains.annotations.NotNull;
+
+/** A panel with the Google App Engine Standard staging flags */
+public class AppEngineStandardStagingPropertiesPanel {
+
+  private JTextField compileEncodingTextField;
+  private JCheckBox deleteJspsCheckBox;
+  private JCheckBox enableJarClassesCheckBox;
+  private JCheckBox disableJarJspsCheckBox;
+  private JCheckBox enableJarSplittingCheckBox;
+  private JCheckBox enableQuickstartCheckBox;
+  private JCheckBox disableUpdateCheckCheckBox;
+  private JTextField jarSplittingExcludesTextField;
+  private JPanel mainPanel;
+
+  public JTextField getCompileEncodingTextField() {
```

Comment:
are these exposed on for testing? if so `@VisibleForTesting`

Reply:
and probably  move them down below the methods that have the core functionality, for readability